### ISSUE TITLE
FederatedSignIn custom provider & custom state

### ISSUE THREAD

**Issue by djerraballi** [aws-amplify/amplify-js#5860]
**Describe the bug**
If a custom provider and a custom state are used in conjunction, amplify incorrectly removes the custom provider and sets the provider to 'COGNITO' in the federated sign in flow. 

https://github.com/aws-amplify/amplify-js/blob/b4b40aa62644ea3688068e4e64f4573c568677a0/packages/auth/src/Auth.ts#L1733-L1736 

**To Reproduce**
Use a custom provider and a custom state together, see that your custom provider will be swapped out in the authorize url to the COGNITO. 

**Expected behavior**

the expectation is that if i am using custom provider, and customState is conditionally present, i shouldnt have to change the input parameter. 

**Code Snippet**

```
Auth.federatedSignIn({
  customProvider: 'some-provider',
  customState: 'something'
})
```

**Comment by djerraballi** [comment]
My work around is in the cases where i need to use customProvider and customState, i change the input param of customProvider to provider. 

if customState is however present, i need to change customProvider back to provider. 

workaround does work, but seems unintentional?

**Comment by manueliglesias** [comment]
Hi @djerraballi 

This is a bug, I am labeling the issue as such, thanks for the report!

And also a big thank you for sharing a workaround for other people that might encounter it.

**Comment by div5yesh** [comment]
This issue is resolved in our latest release 3.0.23.
